libs/num/numeric.go

```diff
@@ -7,12 +7,13 @@ import (
 )
 
 type Numeric struct {
+	asInt     *Int
 	asUint    *Uint
 	asDecimal *Decimal
 }
 
 func (n *Numeric) Clone() *Numeric {
-	if n.asUint != nil {
+	if n.asUint != nil || n.asInt != nil {
 		nn := n
 		return nn
 	}
```

Comment:
that doesn't Clone does it? I think we should be calling the .Clone() function on those types to actualy clone.